pandas/core/groupby/ops.py

```diff
@@ -142,13 +147,13 @@ def _get_group_keys(self):
             # provide "flattened" iterator for multi-group setting
             return get_flattened_iterator(comp_ids, ngroups, self.levels, self.codes)
 
-    def apply(self, f, data, axis: int = 0):
+    def apply(self, f, data: FrameOrSeries, axis: int = 0):
         mutated = self.mutated
         splitter = self._get_splitter(data, axis=axis)
         group_keys = self._get_group_keys()
         result_values = None
 
-        sdata = splitter._get_sorted_data()
+        sdata = splitter._get_sorted_data()  # type: FrameOrSeries
```

Comment:
why is this needed?

shouldn't need to add a type annotation here. maybe the return type of `_get_sorted_data` needs to be added.

Reply:
_get_sorted_data return type is annotated, but mypy complains without this

Reply:
can update to py3.6 syntax in a followon

Reply:
no longer needed after https://github.com/pandas-dev/pandas/pull/28339/commits/e6c5f5a31e5a5912bd572e08f03756cbc075752c